apple/internal/aspects/swift_static_framework_aspect.bzl

```diff
@@ -0,0 +1,152 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Aspect implementation for Swift static framework support."""
+
+load(
+    "@build_bazel_rules_swift//swift:swift.bzl",
+    "SwiftInfo",
+)
+
+SwiftStaticFrameworkInfo = provider(
+    fields = {
+        "module_name": "The module name for the single swift_library dependency.",
+        "swiftinterfaces": """
```

Comment:
I think we might want to propagate swiftmodules as well. That's at least required for using `@testable` with frameworks

Reply:
But I thought the interface files have to be used to deal with different compiler versions, i.e. - this shouldn't be included in something that is to be distributed, and that's the point of this rule, no?

Reply:
Hrm yea good question, maybe the expectation is that you don't use vendored libraries like this with @testable, so you don't need it.

Reply:
Testing the code going into this static framework should go into targets that directly depend on the `swift_library`, so `@testable` in those will work.

If one wanted to write tests on the distributed binary this makes, it should be testing the public interfaces and not going to internal details.

Reply:
I guess it depends on your use case. We currently distribute binary frameworks from libraries internally that have mock code baked in and internal, in those cases we do import the frameworks with `@testable`